Kernel/Process.cpp

```diff
@@ -1551,6 +1556,127 @@ void Process::crash(int signal, u32 eip)
     ASSERT_NOT_REACHED();
 }
 
+void Process::create_core_dump()
+{
+    Elf32_Ehdr elf_file_header;
+    size_t offset = 0;
+
+    sti(); // Re-enable interrupts so we don't shit the bed (we need I/O)!!!
+    auto pname = String(m_name);
+    auto fd_or_error = VFS::the().open(String(String::format("%s_%u.core", pname.characters(), RTC::now())), O_CREAT, 0400, current_directory(), UidAndGid { m_uid, m_gid });
+    auto& fd = fd_or_error.value();
+
+    elf_file_header.e_ident[EI_MAG0] = 0x7f;
+    elf_file_header.e_ident[EI_MAG1] = 'E';
+    elf_file_header.e_ident[EI_MAG2] = 'L';
+    elf_file_header.e_ident[EI_MAG3] = 'F';
+    elf_file_header.e_ident[EI_CLASS] = ELFCLASS32;
+    elf_file_header.e_ident[EI_DATA] = ELFDATA2LSB;
+    elf_file_header.e_ident[EI_VERSION] = EV_CURRENT;
+    elf_file_header.e_ident[EI_OSABI] = 0; // ELFOSABI_NONE
+    elf_file_header.e_ident[EI_ABIVERSION] = 0;
+    elf_file_header.e_ident[EI_PAD + 1] = 0;
+    elf_file_header.e_ident[EI_PAD + 2] = 0;
+    elf_file_header.e_ident[EI_PAD + 3] = 0;
+    elf_file_header.e_ident[EI_PAD + 4] = 0;
+    elf_file_header.e_ident[EI_PAD + 5] = 0;
+    elf_file_header.e_ident[EI_PAD + 6] = 0;
+    elf_file_header.e_ident[EI_NIDENT] = 16;
+    elf_file_header.e_type = ET_CORE;
+    elf_file_header.e_machine = EM_386;
+    elf_file_header.e_version = 1;
+    elf_file_header.e_entry = 0;
+    elf_file_header.e_phoff = sizeof(Elf32_Ehdr);
+    elf_file_header.e_shoff = 0;
+    elf_file_header.e_flags = 0;
+    elf_file_header.e_ehsize = sizeof(Elf32_Ehdr);
+    elf_file_header.e_shentsize = sizeof(Elf32_Shdr);
+    elf_file_header.e_phentsize = sizeof(Elf32_Phdr);
+    elf_file_header.e_phnum = m_regions.size() + 1; // +1 for NOTE section
+    elf_file_header.e_shnum = 0;
+    elf_file_header.e_shstrndx = SHN_UNDEF;
+
+    fd->write(reinterpret_cast<uint8_t*>(&elf_file_header), sizeof(Elf32_Ehdr));
+
+    // TODO: We need to generate appropriate PT_NOTE sections to store
```

Comment:
@ADKaster Do you know how these are supposed to be generated?? The format is really really bizarre and seems to be text based????

Reply:
Well, looking [here] (https://stackoverflow.com/a/38427225) it seems as though what you actually put in the PT_NOTE is not really part of the elf spec? But it looks like Linux at least puts some structures containing the register dump and signal info and generic process info (uptime, user info, crash time etc). I've not looked into that kind of header much yet though.

Reply:
Maybe I should read the PR before commenting :D. You mentioned those things below. Looks like PT_NOTEs are just generic info? But specific ones are used for core dumps on Linux and the BSDs and Solaris. 

Seems there's a special entry in the section header table for SHT_NOTE that points to all the notes, and each note is a type, a name, and a pointer to some offset in the file that contains a note specific structure. https://www.gabriel.urdhr.fr/2015/05/29/core-file/
Looks like the actual structures and "note types" used are extremely OS specific.. I'd be surprised if it was trivial to get gdb to open the core files. Might be more interesting (and hair pulling exciting) to get LibDebug to know how to parse an ET_CORE file. @itamar8910

Reply:
I see. I might have to have a peruse of the GDB source code to see where it tries to read it from then. I imagine there's some kind of tag for it (I think it's NT_PRSTATUS).